sdks/python/apache_beam/transforms/core.py

```diff
@@ -308,14 +308,19 @@ class RunnerAPIPTransformHolder(PTransform):
   without a serialized Python `DoFn` object.
   """
 
-  def __init__(self, proto):
+  def __init__(self, proto, context):
     self._proto = proto
+    self._context = context
 
   def proto(self):
     """Runner API payload for a `PTransform`"""
     return self._proto
 
   def to_runner_api(self, context, has_parts=False):
+    id_to_proto_map = self._context.environments.get_id_to_proto_map()
```

Comment:
This seems a bit fragile in that we're assuming that if the ids match, the protos match (which could be bad for auto-generated names like env0). Could you add a check for this. 

It also seems that we're copying too much (every environment from the context, not just the one(s) referenced from this proto), but perhaps there's no good way to get around that. Could you at least add a TODO referencing the JIRA about making environment a top-level attribute of transforms?

Reply:
Added a check for environments.

Created https://issues.apache.org/jira/browse/BEAM-7850 for making Environment a top level attribute of PTransform.